simplemma/lemmatizer.py

```diff
@@ -151,6 +151,19 @@ def _dehyphen(token: str, datadict: Dict[str, str], greedy: bool) -> Optional[st
     return "".join(splitted)
 
 
+def _find_prefixes(
```

Comment:
I'm a big fan of guard clauses to reduce nesting and simplify the code:

```python
def _find_prefixes(
    token: str, lang: Optional[str], datadict: Dict[str, str]
) -> Optional[str]:
    "Subword decomposition: pre-defined prefixes (often absent from vocabulary if they are not words)."
    if lang not in FIND_KNOWN_PREFIXES:
        return None

    prefix = FIND_KNOWN_PREFIXES[lang](token)
    if prefix is None or len(prefix) >= len(token):
        return None

    subword = _simple_search(token[len(prefix) :], datadict)
    if subword is None:
        return None

    return prefix + subword.lower()
```

I know that it repeats the `return None` more times. But reducing the nesting of if clauses really make things simpler to read

Reply:
OK, I switched to guard clauses.